### ISSUE TITLE
unable to connect to database

### ISSUE THREAD

**Issue by cevich** [ansible/awx#521]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - Installer

##### SUMMARY
Pods created / started, migrations never happen.

##### ENVIRONMENT
* AWX version: HEAD
* AWX install method: openshift, minishift
* Ansible version:  2.4
* Operating System: Fedora 25
* Web Browser: Firefox

##### STEPS TO REPRODUCE
Setup:
-    net.ipv4.ip_forward = 1
-    INSECURE_REGISTRY='--insecure-registry=172.30.0.0/16'
-    open ports 8443, 53, 8053,5432 on zone for docker's bridge interface (172.17.0.0/16)
-    modified ini-file as per docs
-    ran installer as per docs

##### EXPECTED RESULTS

awx-celery should be able to connect to the database

##### ACTUAL RESULTS

```
$ oc logs -f awx-3303925813-csqm4 -c awx-celery
Using /etc/ansible/ansible.cfg as config file
[DEPRECATION WARNING]: The sudo command line option has been deprecated in 
favor of the "become" command line arguments. This feature will be removed in 
version 2.6. Deprecation warnings can be disabled by setting 
deprecation_warnings=False in ansible.cfg.
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: 	TCP/IP connections on port 5432?
127.0.0.1 | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "unable to connect to database: could not connect to server: Connection timed out\n\tIs the server running on host \"postgresql\" (172.30.64.249) and accepting\n\tTCP/IP connections on port 5432?\n"
}
```

...eventual python traceback...

```
$ oc logs -f postgresql-2-wplj9 -c postgresql
waiting for server to start....LOG:  redirecting log output to logging collector process
HINT:  Future log output will appear in directory "pg_log".
 done
server started
ALTER ROLE
waiting for server to shut down.... done
server stopped
LOG:  redirecting log output to logging collector process
HINT:  Future log output will appear in directory "pg_log".
```

##### ADDITIONAL INFORMATION

postgresql deployment config: https://paste.fedoraproject.org/paste/Z52hmN8XNl8ctoi36ToyZQ
awx deployment config: https://paste.fedoraproject.org/paste/gZ7U6uP7kIsIZgS2NT7VLw

**Comment by cevich** [comment]
I tried scaling both pods down to 0, then back up, same errors 😢

**Comment by cevich** [comment]
Messing around, I created a new route and this seems to have fixed the issue.  Here's the YAML for the route:

```yaml
apiVersion: v1
kind: Route
metadata:
  name: postgresql
  namespace: awx
  selfLink: /oapi/v1/namespaces/awx/routes/postgresql
  uid: 11657664-bbef-11e7-8e07-54e1ad07d47a
  resourceVersion: '1155'
  creationTimestamp: '2017-10-28T14:48:32Z'
  annotations:
    openshift.io/host.generated: 'true'
spec:
  host: ##### the default one #####
  to:
    kind: Service
    name: postgresql
    weight: 100
  port:
    targetPort: postgresql
  tls:
    termination: edge
  wildcardPolicy: None
status:
  ingress:
    - host: ##### the default one #####
      routerName: router
      conditions:
        - type: Admitted
          status: 'True'
          lastTransitionTime: '2017-10-28T14:48:32Z'
      wildcardPolicy: None
```

**Comment by cevich** [comment]
Experimenting, I killed off all containers and removed all images (leaving ``/var/lib/origin/`` intact.  Upon re-running:

```
oc cluster up \
        --use-existing-config \
        --host-config-dir=/var/lib/origin/config \
        --host-data-dir /var/lib/origin/data \
        --host-pv-dir=/var/lib/origin/persistent_volumes \
        --host-volumes-dir=/var/lib/origin/openshift_volumes
```

The database-connection error does not reproduce.  Perhaps there is some kind of race-condition WRT providing new storage to the postgresql pod that causes it to fail?

Note: Previously, I did try restarting the pods several times, but always reproduced the originally-reported errors.
